addons/website_blog/controllers/main.py

```diff
@@ -30,7 +30,9 @@ def tags_list(self, tag_ids, current_tag):
         return ','.join(slug(tag) for tag in tag_ids)
 
     def nav_list(self, blog=None):
-        dom = blog and [('blog_id', '=', blog.id)] or []
+        blog = blog or request.params.get('blog')
+        dom = [('blog_id', '=', blog.id)] if blog else []
```

Comment:
We shouldn't have to modify this `nav_list` function by searching for the "blog".
You should rather pass it as a parameter from `_prepare_blog_values`.